modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/DisasterRecoveryManager.java

```diff
@@ -1132,6 +1161,17 @@ private CompletableFuture<Void> processNewRequest(DisasterRecoveryRequest reques
 
         CompletableFuture<Void> operationFuture = new CompletableFuture<Void>().orTimeout(TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
+        CompletableFuture<Void> remoteProcessingFuture = request.type() == DisasterRecoveryRequestType.SINGLE_NODE
```

Comment:
This is hard to read; it needs to be simplified.
Let's make it a separate method and simplify it a bit.